operate/services/manage.py

```diff
@@ -719,48 +732,50 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
 
         # TODO Handle this in a more graceful way.
         agent_id = (
-            staking_params["agent_ids"][0]
-            if staking_params["agent_ids"]
+            target_staking_params["agent_ids"][0]
+            if target_staking_params["agent_ids"]
             else user_params.agent_id
         )
-        staking_params["agent_ids"] = [agent_id]
+        target_staking_params["agent_ids"] = [agent_id]
```

Comment:
I think this can be simplified, no?
target_staking_params already equals the fallback params if no staking.
`agent_id = target_staking_params["agent_ids"][0]`

Reply:
yeah, this can be removed probably. But I'll not touch it for now because we could be missing some edge cases where `user_params.agent_id` is different from the `target_staking_params["agent_ids"][0]`. There is no test coverage also for this